libbeat/logp/file_rotator.go

```diff
@@ -73,16 +75,26 @@ func (rotator *FileRotator) WriteLine(line []byte) error {
 	}
 
 	line = append(line, '\n')
+
+	rotator.currentLock.RLock()
 	_, err := rotator.current.Write(line)
```

Comment:
Yeah you are right, I was trying to keep `Write` under a `RLock` as it may block for a while, and sacrifice some consistency in `currentSize` in favor of performance. Putting everything under a write lock would make goroutines logging wait on others, but yeah, this is probably overkill. What do you think?